contracts/scripts/test/smokeTest.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+trap "exit" INT TERM ERR
+trap "kill 0" EXIT
+nodeWaitTimeout=60
+
+main()  
+{
+    nodeOutput=$(mktemp "${TMPDIR:-/tmp/}$(basename 0).XXX")
+    yarn run node:fork &> $nodeOutput &
+    NODE_PID=$!
+
+    echo "Node output: $nodeOutput"
+    echo "Waiting for node to initialize:"
+    i=0
+    until grep -q -i 'Started HTTP and WebSocket JSON-RPC server at' $nodeOutput
+    do
+      let i++      
+      printf "."
+      sleep 1
+      if (( i > nodeWaitTimeout )); then
+        printf "\n"
+        echo "$newLine Node failed to initialize in $nodeWaitTimeout seconds"
+        exit 1
+      fi
+    done
+    printf "\n"
+    echo "🟢 Node initialized running smoke tests"
+
+    FORK=true npx hardhat smokeTest --network localhost "$@"
+}
```

Comment:
Shouldn't the script kill the NODE_PID otherwise the process will keep running in the background?

As a side node, I don't necessarily see myself needing this script. I'm perfectly fine running the fork node on a terminal (after having set BLOCK_NUMBER env var) and then running the smokeTest task in a separate terminal.

Reply:
The top lines take care of child processes being killed when the main process is terminated: 
```
trap "exit" INT TERM ERR
trap "kill 0" EXIT
```
added that as a comment to this script. 


I created this as a single process script so that we can in future more easily include it into the CI. With the logic that smoke tests are one process running that either exits with 0 (all fine) or an error (something wrong). And that results can be reproducible and consistent between runs.

Reply:
Sounds good!